### DISCUSSION TITLE
Vite templete

### DISCUSSION THREAD

**Discussion by ryota-murakami** [reduxjs/redux-toolkit#1330] [Ideas]
Currently [Create React App redux-toolkit templete](https://redux-toolkit.js.org/introduction/getting-started#installation) is available as a scaffold React Redux project option.

According to [npm trends](https://www.npmtrends.com/create-react-app-vs-vite) [Vite](https://github.com/vitejs/vite) which CRA alternate has become very popular in the last six months.  

<img width="1355" alt="Screen Shot 2021-07-20 at 22 39 12" src="https://user-images.githubusercontent.com/5501268/126335839-54f6f687-dbbc-4e9f-b29d-185b3e184ebe.png">

The main factor is faster build and dev server i think, author's Evan You talked outline of Vite at July, 16.

[**Vite: Rethinking Frontend Tooling**](https://portal.gitnation.org/contents/vite-rethinking-frontend-tooling)  

In the Vite Docs [Community Template](https://vitejs.dev/guide/#community-templates) section, usually template is providing by [degit](https://github.com/Rich-Harris/degit) that simple repository clone cli tool.  

It so written, seems like we can create a Vite templete for the Redux Toolkit without any special effort.  

I think we can provide a more comfortable development environment for users by creating an official Vite template,
what do you think everyone?

**Comment by markerikson** [comment]
Yeah, I'm totally on board with more templates! :)  Like, it would be nice if we had some for React Native and Next as well.

Right now we've got two different template repos, one for JS and one for TS: https://github.com/reduxjs/cra-template-redux and https://github.com/reduxjs/cra-template-redux-typescript .

Maybe it would be worth having a single repo with multiple packages, one for each template?

**Reply by ryota-murakami** [reply]
It's a good easy-to-maintain way. 😀

**Comment by ryota-murakami** [comment]
I happened to find Vite's React RTK Query templete from [awesome Vite](https://github.com/vitejs/awesome-vite).

https://github.com/PrinceRajRoy/vite-react-rtkq

**Comment by ryota-murakami** [comment]
@markerikson Hi, I created Vite templete for Redux-Toolkit!
https://github.com/laststance/vite-redux-toolkit-starter

App code is copy of cra-redux-toolkit template.

I'm glad to try it and give me any feedback while if you have a spare time! 😄

**Reply by markerikson** [reply]
Nice! Yeah, the Redux templates have been kinda neglected for a while - some folks have filed issues or PRs, but I haven't had attention to release anything new.

Tbh I think we ought to consolidate all the templates into a single repo, and then have templates for CRA+JS, CRA+TS, Vite, RN, and Next.  (I know Next has their own `examples` folder, but wouldn't hurt to publish one or something also?)

Could you go file an issue on the current Redux CRA+JS template repo so we can plan this out further?

**Reply by ryota-murakami** [reply]
Sure, I'll submit issue about it!

**Reply by ryota-murakami** [reply]
I don't have a time to taking about this until after a month roughly.  

Fortunately I'll use [NX](https://nx.dev/) that monorepo management tool next paid work project, I hope I'll got better monorepo knowledge through the project!

**Comment by Konstantin-Koktashev** [comment]
I think we need templates for all the major ones, will make adopting rtk into those much easier/quicker.
Templates I think are a must:
NextJs+Typescript+Rtk,
SvelteKit+Typescript+Rtk,
Nuxt+Typescript+Rtk

**Comment by nickserv** [comment]
We now have an officially maintained Vite template: `npx degit reduxjs/redux-templates/packages/vite-template-redux`

https://github.com/reduxjs/redux-templates/tree/master/packages/vite-template-redux

**Comment by ryota-murakami** [comment]
@nickmccurdy Great work nick!
What does [tsconfig.node.json](https://github.com/reduxjs/redux-templates/blob/master/packages/vite-template-redux/tsconfig.node.json) mean?

**Reply by nickserv** [reply]
It's just a Vite convention that configures TS for Node scripts (notably including the Vite config).

**Reply by ryota-murakami** [reply]
@nickmccurdy I see I got it!

**Comment by ryota-murakami** [comment]
I'll Continue to [vite-rtk-query](https://github.com/laststance/vite-rtk-query) project as focusing RTK-query. 😄

**Comment by ryota-murakami** [comment]
I sent 2 PR for Vite Redux templates.

- [[vite-template-redux] fix broken build, type-check by ryota-murakami · Pull Request #103 · reduxjs/redux-templates](https://github.com/reduxjs/redux-templates/pull/103)
- [[vite-template-redux] fix "ESLint couldn't find the plugin "eslint-pulgin-prettier"." by ryota-murakami · Pull Request #102 · reduxjs/redux-templates](https://github.com/reduxjs/redux-templates/pull/102)

And I suggest remove project root configs(`babel.config.js`, `.eslintrc`, .`prettierrc` etc) that makes change each `packages`'s npm scripts behavior.
